random/random_test.go

```diff
@@ -9,119 +9,128 @@ import (
 
 func TestParseDistribution(t *testing.T) {
 	cases := map[string]struct {
-		input string
-		want  random.Distribution
-		ok    bool
+		inputs []string
+		want   random.Distribution
+		ok     bool
 	}{
 		"fixed": {
-			`fixed(1)`,
+			[]string{`fixed(1)`, `fixed:1`},
 			&random.Fixed{Value: 1},
 			true,
 		},
 		"uniform": {
-			`uniform(1..10)`,
+			[]string{`uniform(1..10)`, `uniform:1..10`},
 			&random.Uniform{Min: 1, Max: 10},
 			true,
 		},
 		"invalid fixed #1": {
-			`fixed(asdfs)`,
+			[]string{`fixed(asdfs)`, `fixed:asdfs`},
 			nil,
 			false,
 		},
 		"invalid fixed #2": {
-			`fixed)1(`,
+			[]string{`fixed)1(`},
 			nil,
 			false,
 		},
 		"invalid fixed #3": {
-			`fixed()`,
+			[]string{`fixed()`, `fixed:`},
+			nil,
+			false,
+		},
+		"invalid fixed #4": {
+			[]string{`fixed:1)`},
 			nil,
 			false,
 		},
 		"invalid uniform #1": {
-			`uniform(1..)`,
+			[]string{`uniform(1..)`, `uniform:1..`},
 			nil,
 			false,
 		},
 		"invalid uniform #2": {
-			`uniform(1..)`,
+			[]string{`uniform(..)`, `uniform:..`},
 			nil,
 			false,
 		},
 		"invalid uniform #3": {
-			`uniform(..)`,
+			[]string{`uniform(abc..def)`, `uniform:abc..def`},
 			nil,
 			false,
 		},
 		"invalid uniform #4": {
-			`uniform(abc..def)`,
+			[]string{`uniform(300..100)`, `uniform:300..100`},
 			nil,
 			false,
 		},
 		"invalid uniform #5": {
-			`uniform(300..100)`,
+			[]string{`uniform:10..100)`},
 			nil,
 			false,
 		},
 		"unsupported #1": {
-			`unsupported(1..100, 10)`,
+			[]string{`unsupported(1..100, 10)`, `unsupported:1..100,10`},
 			nil,
 			false,
 		},
 		"unsupported #2": {
-			`~fixed(1..100, 10)`,
+			[]string{`~fixed(1..100, 10)`, `~fixed:1..100,10`},
 			nil,
 			false,
 		},
 	}
 
 	for name, cas := range cases {
 		t.Run(name, func(t *testing.T) {
-			got, err := random.ParseDistribution(cas.input)
-			if err != nil {
+			for _, input := range cas.inputs {
+				got, err := random.ParseDistribution(input)
+				if err != nil {
+					if !cas.ok {
+						return
+					}
+					t.Fatalf("ParseDistribution(%q)=%s", input, err)
+				}
 				if !cas.ok {
-					return
+					t.Errorf("expected ParseDistribution(%q) to fail", input)
+				}
+				if !reflect.DeepEqual(got, cas.want) {
+					t.Errorf("got %+v, want %+v", got, cas.want)
 				}
-				t.Fatalf("ParseDistribution(%q)=%s", cas.input, err)
-			}
-			if !cas.ok {
-				t.Errorf("expected ParseDistribution(%q) to fail", cas.input)
-			}
-			if !reflect.DeepEqual(got, cas.want) {
-				t.Errorf("got %+v, want %+v", got, cas.want)
 			}
 		})
 	}
 }
 
 func TestParseRatio(t *testing.T) {
 	cases := map[string]struct {
```

Comment:
Could we avoid a map here?

Reply:
This is pre-existing code, I think this should be fixed in a separate PR.

Reply:
Sure, @denesb could you please create an issue for that.

Reply:
Out of curiosity, why we want to avoid map?

Reply:
Out of curiosity, why we want to avoid map?

Reply:
I opened https://github.com/scylladb/scylla-bench/issues/17, you can continue the discussion on the issue.

Reply:
And we want exactly the opposite - we want test randomisation, so test-cases do not depend on the order of execution.

Somewhat related - not sure if it landed, but there was a bug for `testing` package to support test shaking to randomise the order. Using maps gives us that for free.

Reply:
You guys should move the discussion to the issue, because I'm not sure it will be possible to link to this discussion later.